### ISSUE TITLE
how do you remove a metric from opentsdb permanently

### ISSUE THREAD

**Issue by gsaray101** [OpenTSDB/opentsdb#536]
there is a metric I am trying to remove from opentsdb permanently. I run this:

./tsdb scan 2013/01/20 avg metric_name  --delete --config=/etc/opentsdb.conf

then I do metasync and tree sync, I still see the metric in my tree and uid table.

Can somebody please explain how do you remove a metric permanently?

**Comment by gsaray101** [comment]
I git clone the opentsdb master and build it. After googling some, I see that someone addressed the problem: https://github.com/jmangs/opentsdb/blob/6cd18f01639a1514e1dbde94373a1a70ece3c95b/src/tools/UidManager.java

However, this does not seem be integrated into the latest master. I performed a diff on this file with the master version, they are different. Is somebody going to comment on this, which one I use? Or is this like trial and error type of solution?

**Comment by manolama** [comment]
Yeah I still plan to pull in @jmangs code but it needs some work to finish it, i.e. a kind of MR job or something to actually go through and clean up the data table, particularly when removing tag keys and values. For metric's Jan's should work so you can use it till we merge it in 2.2.

**Comment by chm052** [comment]
@gsaray101 @manolama Was this ever added to 2.2? I can't see 6cd18f but was there an equivalent change?

**Comment by qingchn** [comment]
I have a same problem,  how can i remove a metric ?  have release  the feature ?

**Comment by laudukang** [comment]
any update?
